modules/_labs/button/react/lib/IconButton.tsx

```diff
@@ -0,0 +1,224 @@
+        },
+        disabled: {
+          background: toggled ? colors.blueberry100 : colors.soap100,
+          icon: toggled ? colors.blueberry300 : colors.soap600,
+        },
+      };
+    case IconButtonVariant.Plain:
+      return {
+        default: {
+          icon: toggled ? colors.blueberry400 : colors.licorice200,
+        },
+        hover: {
+          icon: toggled ? colors.blueberry400 : colors.licorice500,
+        },
+        active: {
+          icon: toggled ? colors.blueberry400 : colors.licorice500,
+        },
+        focus: {
+          icon: toggled ? colors.blueberry400 : colors.licorice500,
+          focusRing: focusRing(2, 0),
+        },
+        disabled: {
+          icon: toggled ? colors.blueberry200 : colors.soap600,
+        },
+      };
+    case IconButtonVariant.Inverse:
+      return {
+        default: {
+          background: toggled ? colors.frenchVanilla100 : undefined,
+          icon: toggled ? colors.blueberry400 : colors.frenchVanilla100,
+        },
+        hover: {
+          background: toggled ? colors.frenchVanilla100 : 'rgba(0, 0, 0, 0.2)',
+          icon: toggled ? colors.blueberry400 : colors.frenchVanilla100,
+        },
+        active: {
+          background: toggled ? colors.frenchVanilla100 : 'rgba(0, 0, 0, 0.3)',
+          icon: toggled ? colors.blueberry400 : colors.frenchVanilla100,
+        },
+        focus: {
+          background: toggled ? colors.frenchVanilla100 : 'rgba(0, 0, 0, 0.2)',
+          icon: toggled ? colors.blueberry400 : colors.frenchVanilla100,
+          focusRing: focusRing(2, 2, true, false, 'currentColor', colors.frenchVanilla100),
+        },
+        disabled: {
+          background: toggled ? 'rgba(255,255,255,0.75)' : 'transparent',
+          icon: toggled ? colors.blueberry400 : 'rgba(255, 255, 255, 0.75)',
+        },
+      };
+    case IconButtonVariant.InverseFilled:
+      return {
+        default: {
+          background: toggled ? colors.frenchVanilla100 : 'rgba(0, 0, 0, 0.2)',
+          icon: toggled ? colors.blueberry400 : colors.frenchVanilla100,
+        },
+        hover: {
+          background: toggled ? colors.frenchVanilla100 : 'rgba(0, 0, 0, 0.3)',
+          icon: toggled ? colors.blueberry400 : colors.frenchVanilla100,
+        },
+        active: {
+          background: toggled ? colors.frenchVanilla100 : 'rgba(0, 0, 0, 0.4)',
+          icon: toggled ? colors.blueberry400 : colors.frenchVanilla100,
+        },
+        focus: {
+          background: toggled ? colors.frenchVanilla100 : 'rgba(0, 0, 0, 0.2)',
+          icon: toggled ? colors.blueberry400 : colors.frenchVanilla100,
+          focusRing: focusRing(2, 2, true, false, 'currentColor', colors.frenchVanilla100),
+        },
+        disabled: {
+          background: toggled ? 'rgba(255,255,255,0.75)' : 'rgba(0, 0, 0, 0.2)',
+          icon: toggled ? colors.blueberry400 : 'rgba(255, 255, 255, 0.75)',
+        },
+      };
+  }
+};
```

Comment:
this is where I was trumped a bunch, mostly around toggled on styles and whatinput styles, did this end up working for all cases?

Reply:
@mannycarrera4 Yeah I rewrote some of the `ButtonContainer` styles to make this work for all cases. You can take a look at the visual testing story if you want to double check 👍